app/javascript/components/profile/avatar-selector/CropFinishedStep.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react'
+import { State, Action } from './reducer'
+import { useMutation } from 'react-query'
+import { typecheck } from '../../../utils/typecheck'
+import { FormButton } from '../../common'
+import { ResultsZone } from '../../ResultsZone'
+import { ErrorBoundary, ErrorMessage } from '../../ErrorBoundary'
+import { User } from '../../types'
+import { camelizeKeys } from 'humps'
+
+type Links = {
+  update: string
+}
+
+const DEFAULT_ERROR = new Error('Unable to upload profile photo')
+
+export const CropFinishedStep = ({
+  state,
+  dispatch,
+  links,
+}: {
+  state: State
+  dispatch: React.Dispatch<Action>
+  links: Links
+}): JSX.Element => {
+  const [submit, { status, error }] = useMutation(
+    () => {
+      if (!state.croppedImage) {
+        throw new Error('Cropped image was expected')
+      }
+
+      const formData = new FormData()
+      formData.append('user[avatar]', state.croppedImage, 'avatar.jpg')
+
+      /* TODO: Use our standard sendRequest library */
```

Comment:
I tried doing this but it ended up in a pretty deep rabbit hole. I've just left this as a TODO for now.